.Lib9c.Tests/Action/MockStateDelta.cs

```diff
@@ -12,157 +15,316 @@ namespace Lib9c.Tests.Action
-        // using appropriate methods such as Transfer/Mint/Burn to set the values.
-        // Also convert to internal data types.
-        public MockStateDelta(
-            IImmutableDictionary<Address, IValue> states = null,
-            IImmutableDictionary<(Address Address, Currency Currency), FungibleAssetValue> balances = null,
-            IImmutableDictionary<Currency, FungibleAssetValue> totalSupplies = null,
-            ValidatorSet validatorSet = null)
+        private MockStateDelta(
+            AccountStateGetter accountStateGetter,
+            AccountBalanceGetter accountBalanceGetter,
+            TotalSupplyGetter totalSupplyGetter,
+            ValidatorSetGetter validatorSetGetter)
         {
-            _states = states ?? ImmutableDictionary<Address, IValue>.Empty;
-            _fungibles = balances is { } b
-                ? b.ToImmutableDictionary(kv => kv.Key, kv => kv.Value.RawValue)
-                : ImmutableDictionary<(Address, Currency), BigInteger>.Empty;
-            _totalSupplies = totalSupplies is { } t
-                ? t.ToImmutableDictionary(kv => kv.Key, kv => kv.Value.RawValue)
-                : ImmutableDictionary<Currency, BigInteger>.Empty;
-            _validatorSet =
-                validatorSet ?? new ValidatorSet();
-
-            _delta = new MockDelta(_states, _fungibles, _totalSupplies, _validatorSet);
+            Delta = new MockDelta();
+            StateGetter = accountStateGetter;
+            BalanceGetter = accountBalanceGetter;
+            TotalSupplyGetter = totalSupplyGetter;
+            ValidatorSetGetter = validatorSetGetter;
+            TotalUpdatedFungibles = ImmutableDictionary<(Address, Currency), BigInteger>.Empty;
         }
 
-        // For Transfer/Mint/Burn
-        private MockStateDelta(
-            IImmutableDictionary<Address, IValue> state,
-            IImmutableDictionary<(Address Address, Currency Currency), BigInteger> balance,
-            IImmutableDictionary<Currency, BigInteger> totalSupplies,
-            ValidatorSet validatorSet)
-        {
-            _states = state;
-            _fungibles = balance;
-            _totalSupplies = totalSupplies;
-            _validatorSet = validatorSet;
+        /// <inheritdoc/>
+        public IAccountDelta Delta { get; private set; }
 
-            _delta = new MockDelta(_states, _fungibles, _totalSupplies, _validatorSet);
-        }
+        /// <inheritdoc/>
+        public IImmutableSet<(Address, Currency)> TotalUpdatedFungibleAssets =>
+            TotalUpdatedFungibles.Keys.ToImmutableHashSet();
 
-        public IAccountDelta Delta => _delta;
+        public IImmutableDictionary<(Address, Currency), BigInteger> TotalUpdatedFungibles
+            { get; protected set; }
 
-        public IImmutableSet<Address> UpdatedAddresses => _delta.UpdatedAddresses;
+        private AccountStateGetter StateGetter { get; set; }
```

Comment:
Is there a reason this private property should be a property? It seems can be a private field.

Reply:
Just copy pasted from libplanet. Blame libplanet I guess (although this is mostly being worked on by me)? 🙄